uniflow/op/model/cv/model_server.py

```diff
@@ -262,3 +268,214 @@ def __call__(self, data: List[str]) -> List[str]:
+            ) from exc
+
+    def _preprocess(self, data: str) -> List[str]:
+        """Preprocess data.
+
+        Args:
+            data (List[str]): Data to preprocess.
+
+        Returns:
+            List[str]: Preprocessed data.
+        """
+        return data
+
+    def _postprocess(self, data: List[str]) -> List[str]:
+        """Postprocess data.
+
+        Args:
+            data (List[str]): Data to postprocess.
+
+        Returns:
+            List[str]: Postprocessed data.
+        """
+        return data
+
+    def __call__(self, data: List[str]) -> List[str]:
+        """Run model.
+
+        Args:
+            data (List[str]): Data to run.
+
+        Returns:
+            List[str]: Output data.
+        """
+        import os
+
+        from open_parser import OpenParser
+
+        cambioml_api_key = os.getenv("CAMBIO_API_KEY")
+        op = OpenParser(cambioml_api_key)
+
+        out = []
+
+        for pdf in data:
+            content_result = op.extract(pdf)
+            out.append(content_result)
+
+        return out
+
+
+class UnstructuredModelServer(AbsModelServer):
+    """Unstructured Model Server"""
+
+    def __init__(
+        self, prompt_template: PromptTemplate, model_config: Dict[str, Any]
+    ) -> None:
+        super().__init__(prompt_template, model_config)
+        self._model_config = UnstructuredModelConfig(**self._model_config)
+
+        try:
+            from unstructured.partition.api import (  # pylint: disable=import-outside-toplevel
+                partition_via_api,
+            )
+
+        except ModuleNotFoundError as exc:
+            raise ModuleNotFoundError(
+                "Please install unstructured to use UnstructuredModelServer. You can use `pip install unstructured` to install it."
+            ) from exc
+
+    def _preprocess(self, data: str) -> List[str]:
+        """Preprocess data.
+
+        Args:
+            data (List[str]): Data to preprocess.
+
+        Returns:
+            List[str]: Preprocessed data.
+        """
+        return data
+
+    def _postprocess(self, data: List[str]) -> List[str]:
+        """Postprocess data.
+
+        Args:
+            data (List[str]): Data to postprocess.
+
+        Returns:
+            List[str]: Postprocessed data.
+        """
+        return data
+
+    def __call__(self, data: List[str]) -> List[str]:
+        """Run model.
+
+        Args:
+            data (List[str]): Data to run.
+
+        Returns:
+            List[str]: Output data.
+        """
+        import os
+
+        from unstructured.partition.api import partition_via_api
+
+        out = []
+
+        for pdf in data:
+            elements = partition_via_api(
+                filename=pdf,
+                api_key=os.getenv("UNSTRUCTURED_API_KEY"),
+                api_url=os.getenv("UNSTRUCTURED_API_URL"),
+            )
+            out.append("\n\n".join([str(el) for el in elements]))
```

Comment:
qq: why we need \n\n here?

Reply:
is because unstructured doesn't return in md format. It returns text with something like <unstructured.documents.title> or <unstructured.documents.content>

Reply:
ic. what is the elements concept here?